src/parser/jobs/sam/modules/ThirdEye.js

```diff
@@ -0,0 +1,86 @@
+import {ActionLink} from 'components/ui/DbLink'
+import React from 'react'
+import {Trans, i18nMark} from '@lingui/react'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+export default class ThirdEye extends Module {
+	static handle = 'Third Eye'
+	static i18n_id = i18nMark('sam.thirdeye.title')
+	static dependencies = [
+		'suggestions',
+	]
+
+	_thirdEyes = 0 //this is the amount of times third eye is used
+	_openEyes = 0 //this is the amount of times third eye was proc'ed
+	_spentEyes = 0 //this is the amount of times they used a proc
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {by: 'player'}, this._onCast)
```

Comment:
It'd be cool to get this callback handled as a few separate hooks rather than the one, you should be able to filter the ability ID on each. Not blocking.